indico/queries/model_groups/metrics.py

```diff
@@ -98,11 +98,20 @@ def __init__(self, model_group_id: int):
         super().__init__(self.query, variables={"modelGroupId": model_group_id})
 
     def process_response(self, response):
-        return json.loads(
-            super().process_response(response)["modelGroups"]["modelGroups"][0][
-                "selectedModel"
-            ]["evaluation"]["metrics"]
-        )
+        try:
```

Comment:
If there are no responses returned, graphql should return `[]` or something similar. I think a cleaner way would be to do that and then if no model group is found matching the id, then raise the error instead of catching index error. Index error is on `model_groups[0]` but it might be obscured by catching any index error in the entire block

Reply:
Object returns like: 

`{'data': {'modelGroups': {'modelGroups': []}}}`

Reply:
to me checking that [] empty vs. indexerror (which would always be that it is empty), same... but happy to change if you prefer that

Reply:
so that endpoint filters instead of returning an error if not found same way it returns nothing if filtering by dataset id and no model groups are found. I think the logic below is fine or could be done with out try except by checking len and that the model group id matches. Instead of trusting the indexing error.

Long term we will make a direct way to get model groups that would offer errors for missing items but I think this patch is functional for now.